pkg/destroy/aws/aws.go

```diff
@@ -2021,3 +2079,54 @@ func removeSharedTag(ctx context.Context, tagClients []*resourcegroupstaggingapi
 
 	return nil
 }
+
+// destroyedResources stores the results of attempts to destroy resources. The key is the ARN of the resource. The value
+// is whether the resource was successfully destroyed.
+type destroyedResources map[string]bool
+
+// record records where a resource was successfully destroyed. If the specified error is nil, then the resource was
+// destroyed successfully.
+func (r destroyedResources) record(arn string, err error) {
+	r[arn] = (err == nil)
+}
+
+// merge merges the record of destroyed resources from changes into r. Any entries that are true in either r or changes
+// will be set to true in r.
+func (r destroyedResources) merge(changes destroyedResources) {
+	for k, v := range changes {
+		r[k] = r[k] || v
+	}
+}
+
+// extractRemaining removes the resources that could not be successfully deleted from r and returns a slice containing
+// the ARNs of those resources.
+func (r destroyedResources) extractRemaining() []string {
+	if r == nil {
+		return nil
+	}
+	var notDestroyed []string
+	for k, v := range r {
+		if !v {
+			notDestroyed = append(notDestroyed, k)
+			delete(r, k)
+		}
+	}
+	return notDestroyed
+}
+
+// arnString returns the ARN, as a string, for a resource.
+// client is the client used to access the resource. For example, if the resource is a VPC, then the client would be
+// an EC2 client.
+func arnString(client *awsclient.Client, includeRegion bool, accountID, resourceType, resourceID string) string {
+	var region string
+	if includeRegion {
+		region = *client.Config.Region
+	}
+	return arn.ARN{
+		Partition: client.PartitionID,
+		Service:   client.ServiceName,
+		Region:    region,
+		AccountID: accountID,
+		Resource:  fmt.Sprintf("%s/%s", resourceType, resourceID),
+	}.String()
+}
```

Comment:
I think this function is trying to overly estimate the ARN for a resourceType and resourceId, I am not really confident that this is a good assumption that we should be making. I think currently the only place we create ARN is when we STOP instances for the cluster but that is one resource and this PR adds quite a bit more.